chrahunt/TagProReplays
# Refactor background page replay handling.

Recently a change was made to bring handling of the details of the replay
data to the background page. This builds on that to consolidate and hide
replay metadata generation/usage from most of the background page.

Fixes #157. We now only grab the replays themselves when existing meta-
data cannot be found. This seems to speed up menu load a good amount but
more testing with larger numbers of replays is needed. For that, #57
should be in place first otherwise it will take a while.

Also fixes #158.

Testing done:
1. Record 10 replays across 3 actual games
2. should be able to successfully import 40 replays
3. should import a single replay known to be invalid (#validation) -- result: a dialog box appears stating the replay is invalid along with more information
4. import several replays where one or two are known to be invalid (#validation) -- result: when the invalid replays are encountered rendering should pause until the user dismisses the dialog and then should continue
5. export 1 - validate the text content
6. export 5 - validate the zip
7. Render a short replay and make sure it looks ok
    - Make sure the menu reflects the rendering status in real time
    - When the rendering is finished it should end with a check
8. Render multiple short replays and ensure they do the above in sequence.
9. Render a long replay and ensure that no error occurs
10. On menu refresh, the rendered status of the known-to-be-rendered replay should be accurate
11. The downloaded movie should be playable and the same one that was clicked on in the menu
    - Replay should be deleted, removed from menu, and not come back on refresh.
        - Metadata should be removed from the background page local storage
        - No error should be present on the background page
        - The rendered file, if any, should be removed from the filesystem - set a breakpoint and check the rendered_ids on menu request.
12. make sure sorting works with replays on the relevant facets
13. rename replay from the main menu, the relevant line should be replaced with the new name and it should persist on refresh
14. It should be possible to preview a replay
15. preview an imported replay
16. preview a saved replay
17. Deletion of a replay should work the same as deletion of the single replay.
18. Cropping (Save As) replay
19. Cropping and replacing replay
    - with new name
    - with same name
20. Renaming replay in previewer
21. Recorded replay should trim off beginning from pre-game
